src/accelerate/commands/estimate.py

```diff
@@ -212,6 +212,40 @@ def estimate_command_parser(subparsers=None):
     return parser
 
 
+def estimate_training_usage(bytes: int, mixed_precision: str, msamp_config: str = None) -> dict:
+    """
+    Given an amount of `bytes` and `mixed_precision`, calculates how much training memory is needed for a batch size of
+    1.
+
+    Args:
+        bytes (`int`):
+            The size of the model being trained.
+        mixed_precision (`str`):
+            The mixed precision that would be ran.
+        msamp_config (`str`):
+            The msamp config to estimate the training memory for if `mixed_precision` is set to `"fp8"`.
+    """
+    memory_sizes = {"model": -1, "optimizer": -1, "gradients": -1, "step": -1}
+    fp32_size = bytes
+    fp16_size = bytes // 2
+
+    if mixed_precision == "float32":
+        memory_sizes["model"] = fp32_size
+        memory_sizes["gradients"] = fp32_size
+        memory_sizes["optimizer"] = fp32_size * 2
+        memory_sizes["step"] = fp32_size * 4
+    elif mixed_precision in ("float16", "bfloat16") or mixed_precision == "fp8" and msamp_config is None:
+        # With mixed precision training, the model has weights stored
+        # in FP16 and FP32
```

Comment:
I think that the FP8 case is not handled properly. WE should introduce `fp8_size` maybe . Also I think it is missing parentheses around `mixed_precision == "fp8" and msamp_config is None `.

Reply:
There is no real memory savings with FP8 rn, it is the exact same as with float16 :)